### PR TITLE
Change repr methods to avoid Layer materialization

### PR DESCRIPTION
**Pull request by rjzamora**
The DataFrame and Array `repr`s currently include information about the total number of tasks in the collection's underlying graph. It makes a lot of sense that this information was included 5+ years ago. However, now that graph is always a  `HighLevelGraph` object for DataFrame and Array collections, it probably makes sense to display the total number of HLG `Layers` rather than the total number of **low-level** tasks.  For example:

```python
timeseries(end='2000-01-03').shuffle("id", shuffle="tasks")["id"]
```
```
Dask Series Structure:
npartitions=2
    int64
      ...
      ...
Name: id, dtype: int64
Dask Name: getitem, 3 layers
```

The problem with the current approach (displaying the total number of tasks) is that calling `len` on some layers (like shuffle layers) will result in pre-mature materialization.  For example:

```python
s = timeseries(end='2000-01-03').shuffle("id", shuffle="tasks")["id"]
def check(dsk):
    assert all([not l.is_materialized() for l in s.dask.layers.values()])
check(s.dask)  # Passes
repr(s)
check(s.dask)  # Fails
```

**Further Background and Notes**

Although the `Layer`-materialization problem can be solved by implementing a "smart" `__len__` method for all layers, I feel pretty strongly that we should move in the opposite direction of that kind of solution.  That is, in the spirit of #9216, I think we would benefit from a much simpler `Layer` design that does **not** inherit from `Mapping` at all (and is therefore much easier to implement and much more difficult to inadvertently materialize). For example, I don't even think it should be possible to get the length of a `Layer`, because you should need to call `layer.subgraph()` to explicitly convert the `Layer` object to a proper `Mapping`.  In the same vein, it would be much safer to define a stricter boundary between the HLG and the low-level graph.  For example, perhaps `dict(HighLevelGraph)` should just be synonymous with `HighLevelGraph.layers`, and a distinct `HighLevelGraph.materialize()` method should be required to construct a low-level graph?

### CODE REVIEW

**Comment by rjzamora** [comment]
Note that @jrbourbeau and @jsignell mentioned offline that they would probably be in favor of removing the task information from the `repr` without replacing it with anything (e.g. layer count), since this information is "pre-optimization" anyway.

**Comment by ncclementi** [comment]
How does this affect the recommendations we have in Best Practices about "avoid too large task graphs"? We usually say that graphs that have more than 10000 or 100000 tasks are too big. See this [blog](https://blog.dask.org/2021/11/02/choosing-dask-chunk-sizes)

Is this something that changes when talking about HLG? should we modify the docs to be consistent, or at least mention how to still be able to access the task number? 

Completely understandable if these questions are out of scope. We can move this discussion to a new issue after this gets merged.

File: dask/dataframe/tests/test_format.py [hunk 1]
```diff
@@ -567,7 +567,7 @@ def test_empty_repr():
 <div>
 {style}{exp_table}
 </div>
-<div>Dask Name: from_pandas, 1 tasks</div>""".format(
+<div>Dask Name: from_pandas, 1 layers</div>""".format(
```
[the remark this answers is not in the document]
Comment (rjzamora) [reply dask/dataframe/tests/test_format.py]: good call - I added this in my latest commit

**Comment by rjzamora** [comment]
I haven't removed the `Layer`-count information from the `repr` just yet, because I agree with @ian-r-rose that having this kind of information can be nice (and accessing the layer count is fast/easy).  That said, I am now a bit worried that replacing `"{num_tasks} tasks"` with `"{num_layers} layers"` may lead to some confusion for `dask.array` users that are not familiar with the HLG/Layer system (i.e. **most users**).  For example, the typical user may assume that "2 layers" in the `repr` below has something to do with the structure of the Array:

Screen Shot 2022-07-21 at 10 13 46 AM

In reality, the number of layers roughly corresponds to the number of global (lazy) operations that have been performed on the array, but saying `2 operations` may also be slightly wrong and confusing.  Perhaps we should try to come up with a concise way to convey what 'layers' are here?  Maybe something like `2 graph layers`, `2 task layers`, `2 compute layers`, ... ?

**Comment by jsignell** [comment]
That whole table doesn't make a ton of sense if you think about it too hard. Like `float64` isn't the `type` of the Dask array. I think it's fine to just change `Tasks` to `Graph Layers`. Another options might be to try to include the visual representation of the HLG somehow. Probably that would look like some type of dropdown.

**Comment by rjzamora** [comment]
Sounds good @jsignell - Thanks for taking a look at this!  I agree that we probably want to revise the repr altogether, but I just changed `''<n> layers"` to `"<n> graph layers"` for now.

**Review by ian-r-rose — APPROVED** [review]
Thanks @rjzamora! This looks good to me, and removes a painful failure mode

**Pull request merged**